### PR TITLE
[New] Added oauth2 userinfo endpoint

### PR DESCRIPTION
**Pull request by geekgonecrazy**
@RocketChat/core 

Added as per part of the spec: https://connect2id.com/products/server/docs/api/userinfo

Ran into an application that needed this on the server to finish the oauth login process. Since its part of the spec, I went ahead and added it.

### CODE REVIEW

File: packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee [hunk 1]
```diff
@@ -8,6 +8,27 @@ oauth2server = new OAuth2Server
 
 WebApp.connectHandlers.use oauth2server.app
 
+oauth2server.routes.get '/oauth/userinfo', (req, res, next) ->
+	if not req.headers.authorization?
+		return res.sendStatus(401).send('No token')
+
+	accessToken = req.headers.authorization.replace('Bearer ', '')
+
+	token = oauth2server.oauth.model.AccessTokens.findOne accessToken: accessToken
```
Comment (rodrigok) [inline packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee]: The idea is return an exception if the token was not found?

File: packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee [hunk 2]
```diff
@@ -8,6 +8,27 @@ oauth2server = new OAuth2Server
...
+
+	user = RocketChat.models.Users.findOneById(token.userId);
```
Comment (rodrigok) [inline packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee]: Same thing here?

File: packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee (hunk 1 above)
Comment (geekgonecrazy) [inline packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee]: Added checks on this and the user.

File: packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee (hunk 2 above)
Comment (geekgonecrazy) [inline packages/rocketchat-oauth2-server-config/oauth/server/oauth2-server.coffee]: Added a check here as well

**Pull request merged**